### PR TITLE
Create unit tests

### PR DESCRIPTION
**Pull request by SvenSchiffner**

### CODE REVIEW

File: src/adapt.c [hunk 1]
```diff
@@ -146,6 +146,11 @@ int adapt_open()
   if (!config_read_file(&cfg, file_name))
   {
     fprintf(error_stream, "reading config file %s failed\n", file_name);
+    fprintf(error_stream, "Informations from libconfig:\n");
+    fprintf(error_stream, "File: %d\n", config_error_type(&cfg));
```
Comment (rschoene) [inline src/adapt.c]: Type not File

File: knobs/dct.c [hunk 2]
```diff
@@ -29,10 +29,10 @@
 #include <stdlib.h>
 #include <dlfcn.h>
 
-/*#define VERBOSE 1 */
+#define VERBOSE 1
```
Comment (rschoene) [inline knobs/dct.c]: do not define verbose in code

File: tests/prepare_new_libconfig_for_travis.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (rschoene) [inline tests/prepare_new_libconfig_for_travis.sh]: please note why other versions of libconfig are broken

**Review by rschoene — CHANGES_REQUESTED** [review]

File: knobs/dct.c (hunk 2 above)
Comment (SvenSchiffner) [reply knobs/dct.c]: removed in newer commit

**Review by rschoene — APPROVED** [review]

**Pull request merged**